core/src/main/java/hudson/model/Result.java

```diff
@@ -195,9 +195,6 @@ public OptionHandlerImpl(CmdLineParser parser, OptionDef option, Setter<? super
         public int parseArguments(Parameters params) throws CmdLineException {
             String param = params.getParameter(0);
             Result v = fromString(param.replace('-', '_'));
-            if (v==null)
```

Comment:
Looks like there is a regression introduced at some point.
This method will now accept any argument and fallback to `FAILURE`. So a user can pass `SUCESS` in the CLI argument and get a zero code with unexpected result.

Reply:
Thanks, you are right. I replaced it with a check for FAILURE instead of null.